packages/EasyPagination/src/Paginators/DoctrineDbalLengthAwarePaginator.php

```diff
@@ -64,7 +64,14 @@ protected function doGetResult($queryBuilder): array
      */
     protected function doGetTotalItems($queryBuilder, string $countAlias): int
     {
-        $result = (array)$this->conn->fetchAssoc($queryBuilder->getSQL(), $queryBuilder->getParameters());
+        // @TODO remove this method_exists check when doctrine/dbal is updated to a version containing the fetchAssociative method (>=2.11)
+        $result = \method_exists($this->conn, 'fetchAssociative')
+            ? (array)$this->conn->fetchAssociative(
+                $queryBuilder->getSQL(),
+                $queryBuilder->getParameters(),
+                $queryBuilder->getParameterTypes()
+            )
+            : (array)$this->conn->fetchAssoc($queryBuilder->getSQL(), $queryBuilder->getParameters());
```

Comment:
I think we can pass the same set of arguments to both `fetchAssociative` and `fetchAssoc` methods.
If so, it can be rewritten like so:
```php
$methodName = \method_esists($this->conn, 'fetchAssociative') ? 'fetchAssociative' : 'fetchAssoc';
$result = (array) $this->conn->$methodName(...);
```
As for me, it seems a bit readable. What do you think?

Reply:
Agreed, fixed

Reply:
Agreed, fixed https://github.com/eonx-com/easy-monorepo/pull/836/commits/faa856097d12469d2170a4ce9b98c4ea51503368